### ISSUE TITLE
If given garbage SSL credentials data a Python exception should be raised to the application

### ISSUE THREAD

**Issue by nathanielmanistaatgoogle** [grpc/grpc#467]
The server shouldn't ever segfault.

If given nonsense garbage credentials, it should probably report that to its users at the time that the credentials are passed (currently server creation time, though I think that's slated to change?).

**Comment by ctiller** [comment]
@nathanielmanistaatgoogle - can you provide an example that causes this crash?
(we can probably shift it from @jboeuf's plate then)

**Comment by nathanielmanistaatgoogle** [comment]
Back in February when I reported this issue I was able to make the core crash by providing "this bytestring is not a legitimate private key" as the data of a private key and "this bytestring is not a legitimate certificate chain" as the data of a certificate chain. Reattempting today I am no longer able to reproduce a crash of the core and I see "Handshaker factory creation failed with TSI_INVALID_ARGUMENT" and "Unable to create secure server with credentials of type Ssl" in my log.

I no longer see any behavior that I'd consider Beta-milestone-worthy. I'd like to keep this issue open without a target milestone because the ideal outcome for passing garbage credential data would be a Python exception raised to the application (although whether at credential-creation, add-port, or server-start time I don't yet know).

Is there a core API method return code indicating misbehavior on my code's part that I'm likely ignoring? The logging messages are nice but have we missed something programmatically in place that would indicate our credential data being invalid and the server not able to operate as we intend?
